### PR TITLE
rc: salamander

### PR DESCRIPTION
**Pull request by arboleya**
- [x] #1310
 - [x] #1374
 - [x] #1403
 - [x] #1426
 - [x] #1445
 - [x] #1467
 - [x] #1474
 - [x] #1496
 - [x] #1497
 - [x] #1520
 - [x] #1529
 - [x] #1544
 - [x] #1559
 - [x] #1566
 - [x] #1567
 - [x] #1574
 - [x] #1575
 - [x] #1579
 - [x] #1580
 - [x] #1582
 - [x] #1583
 - [x] #1586
 - [x] #1590
 - [x] #1591
 - [x] #1594
 - [x] #1601
 - [x] #1602
 - [x] #1614
 - [x] #1615
 - [x] #1619
 - [x] #1629
 - [x] #1630
 - [x] #1637
 - [x] #1640
 - [x] #1642
 - [x] #1648
 - [x] #1654
 - [x] #1657
 - [x] #1658
 - [x] #1659
 - [x] #1660
 - [x] #1661
 - [x] #1676
 - [x] #1679

### CODE REVIEW

File: .github/workflows/rc-release.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          published_version=$(echo "$changetsets" | grep -oP '@\K([0-9]+\.){2}[0-9]+-${{ steps.rc_name.outputs.replaced }}-\d+' | head -1)
+          echo "published_version=$published_version" >> $GITHUB_OUTPUT
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: mshick/add-pr-comment@v2
+        with:
+          message: |
+            This RC is published in NPM with version **${{ steps.rc_name.outputs.replaced }}**
```
Comment (arboleya) [inline .github/workflows/rc-release.yaml:56]: @danielbate I was thinking we could maybe improve this message.

What do you think?


```suggestion
            RC published under the `**${{ steps.rc_name.outputs.replaced }}**` tag.
            
            Install it:

            ```bash
            pnpm add fuels@{{ steps.rc_name.outputs.replaced }}
            pnpm add fuels@0.0.0-rc-salamander-20231215132700 # make it dynamic?
            ```

            Check it out:
             - https://www.npmjs.com/package/fuels/v/0.0.0-rc-salamander-20231215132700
```

Comment (arboleya) [reply .github/workflows/rc-release.yaml:56]: Addressed via:
 - https://github.com/FuelLabs/fuels-ts/pull/1559

**Comment by Torres-ssf** [comment]
It seems we have a flaky test on Salamander RC: https://github.com/FuelLabs/fuels-ts/actions/runs/7301835259/job/19899474795#step:9:329

**Comment by Torres-ssf** [comment]
> It seems we have a flaky test on Salamander RC: https://github.com/FuelLabs/fuels-ts/actions/runs/7301835259/job/19899474795#step:9:329
> 
> UPDATE: It is also happening locally, but not always.
> 
> Screenshot 2023-12-22 at 13 48 26

Are we running tests concurrently? A possible case for this one to be flaky is that some additional(s) block(s) may be produced or not during this test.

**Comment by danielbate** [comment]
Yes, the Vitest migration in #1310 means a lot of tests run concurrently, and differently to how they were with jest, hence the performance improvements. I'll look into isolating this test.

**Review by Torres-ssf — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**